From: Matt Fleming <matt.fleming@intel.com>
Date: Tue, 28 Feb 2012 13:37:23 +0000
Subject: [PATCH] x86, boot: Restrict CFLAGS for hostprogs

Git-commit: d40f833630a1299fd377408dc8d8fac370d621b0
Patch-mainline: v3.4
References: fate#313640, fate#314486
Target: SLE-11 SP3

Currently tools/build has access to all the kernel headers in
$(srctree). This is unnecessary and could potentially allow
tools/build to erroneously include kernel headers when it should only
be including userspace-exported headers.

Unfortunately, mkcpustr still needs access to some of the asm kernel
headers, so explicitly special case that hostprog.

Joey Lee:
Replace include/linux/kconfig.h with include/generated/autoconf.h for
compiler on v3.0 kernel, it can avoid to backport more kconfig patches
from v3.1 kernel.

Cc: H. Peter Anvin <hpa@zytor.com>
Signed-off-by: Matt Fleming <matt.fleming@intel.com>
Link: http://lkml.kernel.org/r/1330436245-24875-5-git-send-email-matt@console-pimps.org
Signed-off-by: H. Peter Anvin <hpa@zytor.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
---
 arch/x86/boot/Makefile |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/arch/x86/boot/Makefile
+++ b/arch/x86/boot/Makefile
@@ -44,8 +44,9 @@ setup-y		+= video-bios.o
 targets		+= $(setup-y)
 hostprogs-y	:= mkcpustr tools/build
 
-HOST_EXTRACFLAGS += $(LINUXINCLUDE)
-
+HOSTCFLAGS_mkcpustr.o := -I$(srctree)/arch/$(SRCARCH)/include
+HOST_EXTRACFLAGS += -I$(objtree)/include -I$(srctree)/tools/include \
+                   -include include/generated/autoconf.h
 $(obj)/cpu.o: $(obj)/cpustr.h
 
 quiet_cmd_cpustr = CPUSTR  $@
